### PR TITLE
New Empty States

### LINKED ISSUE #3310: New Empty states
# Current Behavior

The empty states consist of two texts and a call-to-action (CTA):

| 404 & 500 Error pages | Indexes Empty state |
| --- | --- |
| image | image |

# Desired behavior

We want to update the following empty States with an animated illustration:

**ERROR PAGES**

- [500 error page](https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=19450-93391&mode=design&t=OsVwULzMocuuEGia-11)
- [404 error page](https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=19450-93376&mode=design&t=OsVwULzMocuuEGia-11)

**INDEXES**

- [No records on an index](https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=19429-87978&mode=design&t=OsVwULzMocuuEGia-11)
- [500 error on an index](https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=19439-92929&mode=design&t=OsVwULzMocuuEGia-11)

**SHOWS**

-[ No activities in the record timeline.](https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=19429-88646&mode=design&t=OsVwULzMocuuEGia-11)
- [No tasks associated with the record.](https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=19429-89899&mode=design&t=OsVwULzMocuuEGia-11)
- [No notes associated with the record.](https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=19429-90524&mode=design&t=OsVwULzMocuuEGia-11)
- [No files associated with the record.](https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=19437-91183&mode=design&t=OsVwULzMocuuEGia-11)

The goal is to make the illustration track the mouse:

https://github.com/twentyhq/twenty/assets/19412894/edb1624b-30d6-41a2-bb3a-1b246918cb45

Ensure that the size of the Figma files is appropriate. For small illustrations, the largest side should be 160px, while for larger illustrations like error 500 and 404, the bigger side should be 240px.

image

### Export the images:

Export the images in PNG format using the X2 parameter in Figma:

1. Select an image.
2. Set up the export preferences.
3. Click "export."

CleanShot 2024-01-08 at 17 09 23

### Implement the effect.

Please ensure that the dimensions of the illustrations are as specified above.

Bonus: The moving image should return to its initial position when the mouse leaves the window. Use Framer Motion as library.

[👀See effect on codesandbox
](https://codesandbox.io/p/sandbox/moving-effect-twenty-empty-states-pgzcc2?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clr5039vi00063b6k99w7fxz9%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clr5039vi00023b6k8m8eto64%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clr5039vi00033b6k87r4hc3g%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clr5039vi00053b6ktyz6bwwm%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B58.21267403811194%252C41.78732596188806%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clr5039vi00023b6k8m8eto64%2522%253A%257B%2522id%2522%253A%2522clr5039vi00023b6k8m8eto64%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clr5039vi00053b6ktyz6bwwm%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr5039vi00043b6kqu8l09x2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clr5039vi00053b6ktyz6bwwm%2522%252C%2522activeTabId%2522%253A%2522clr5039vi00043b6kqu8l09x2%2522%257D%252C%2522clr5039vi00033b6k87r4hc3g%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clr5039vi00033b6k87r4hc3g%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Implement the new titles & subtiles

Screenshot 2024-01-08 at 17 25 22

(The spacing also have been updated)

# Figma

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=19450-97860&mode=design&t=OsVwULzMocuuEGia-11

# Codesandbox

[Link](https://codesandbox.io/p/sandbox/moving-effect-twenty-empty-states-pgzcc2?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clr5039vi00063b6k99w7fxz9%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clr5039vi00023b6k8m8eto64%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clr5039vi00033b6k87r4hc3g%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clr5039vi00053b6ktyz6bwwm%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B58.21267403811194%252C41.78732596188806%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clr5039vi00023b6k8m8eto64%2522%253A%257B%2522id%2522%253A%2522clr5039vi00023b6k8m8eto64%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clr5039vi00053b6ktyz6bwwm%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr5039vi00043b6kqu8l09x2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clr5039vi00053b6ktyz6bwwm%2522%252C%2522activeTabId%2522%253A%2522clr5039vi00043b6kqu8l09x2%2522%257D%252C%2522clr5039vi00033b6k87r4hc3g%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clr5039vi00033b6k87r4hc3g%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### PR DESCRIPTION
**Pull request by Kanav-Arora**
Closes #3310

### CODE REVIEW

**Comment by Kanav-Arora** [comment]
Hi @lucasbordeau 
Please review this PR

File: packages/twenty-front/src/modules/ui/layout/animated-placeholder/components/EmptyPlaceholderStyles.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  flex-direction: column;
+  gap: ${({ theme }) => theme.spacing(3)};
+  justify-content: center;
+  text-align: center;
+  width: 100%;
+`;
+
+export const StyledEmptyTitle = styled.div`
+  color: ${({ theme }) => theme.font.color.primary};
+  font-size: ${({ theme }) => theme.font.size.md};
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/ui/layout/animated-placeholder/components/EmptyPlaceholderStyles.tsx:26]: Should be `lg` to match Figma

File: packages/twenty-front/src/modules/ui/layout/animated-placeholder/components/EmptyPlaceholderStyles.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  font-weight: ${({ theme }) => theme.font.weight.semiBold};
+  line-height: ${({ theme }) => theme.text.lineHeight.md};
+`;
+
+export const StyledEmptySubTitle = styled.div`
+  color: ${({ theme }) => theme.font.color.tertiary};
+  font-size: ${({ theme }) => theme.font.size.xs};
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/ui/layout/animated-placeholder/components/EmptyPlaceholderStyles.tsx:33]: Should be sm

**Review by lucasbordeau — COMMENTED** [review]
Bug on mobile display : 


https://github.com/twentyhq/twenty/assets/26528466/7162c57b-170d-483d-922e-41b4d9f8194c

**Comment by Kanav-Arora** [comment]
Hi @lucasbordeau 
Done with the changed I have also fixed the mobile bug
Please review

**Comment by Kanav-Arora** [comment]
Hi @lucasbordeau 
Done

**Comment by Kanav-Arora** [comment]
Hi @lucasbordeau 
Please review this

**Review by lucasbordeau — APPROVED** [review]

**Comment by lucasbordeau** [comment]
@Kanav-Arora Nice work !

**Pull request merged**